komodo/cli.py

```diff
@@ -193,57 +193,156 @@ def cli_main():
         add_help=False,
     )
 
-    parser.add_argument("pkgs", type=YamlFile())
-    parser.add_argument("repo", type=YamlFile())
+    parser.add_argument(
+        "pkgs",
+        type=YamlFile(),
+        help="A Komodo release file mapping package name to version, "
+        "in YAML format.",
+    )
+    parser.add_argument(
+        "repo", type=YamlFile(), help="A Komodo repository file, in YAML format."
+    )
 
     required_args = parser.add_argument_group("required named arguments")
 
-    required_args.add_argument("--prefix", "-p", type=str, required=True)
-    required_args.add_argument("--release", "-r", type=str, required=True)
+    required_args.add_argument(
+        "--prefix",
+        "-p",
+        type=str,
+        required=True,
+        help="The path of the directory in which you would like to place the "
+        "built environment.",
+    )
+    required_args.add_argument(
+        "--release",
+        "-r",
+        type=str,
+        required=True,
+        help="The name of the release, will be used as the name of the directory "
+        "containing the enable script and environment `root` directory.",
+    )
     required_args.add_argument(
         "--locations-config",
         type=str,
         required=True,
-        help="Path to locations.yml, a map of location to a server.",
+        help="Path to locations file, mapping locations to servers, " "in YAML format.",
     )
 
     optional_args = parser.add_argument_group("optional arguments")
 
     optional_args.add_argument(
-        "--help", "-h", action="help", help="show this help message and exit"
+        "--help",
+        "-h",
+        action="help",
+        help="Show this help message and exit.",
+    )
+    optional_args.add_argument(
+        "--tmp",
+        "-t",
+        type=str,
+        help="The directory to use for builds by cmake. Defaults to "
+        "current working directory.",
+    )
+    optional_args.add_argument(
+        "--cache",
+        "-c",
+        type=str,
+        default="pip-cache",
+        help="The temporary directory used for downloads, e.g. by pip.",
+    )
+    optional_args.add_argument(
+        "--jobs",
+        "-j",
+        type=int,
+        default=1,
+        help="The number of parallel jobs to use for builds by cmake. Default: 1.",
```

Comment:
By using `parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter) `the defaults are automatically injected into the help text.

Reply:
That is very cool, I should have thought to look for it. Fixed this in the other PR (`cli-help-strings`)